### PR TITLE
School Dance editorial

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/6_Advanced/Max_Flow.mdx [hunk 1]
```diff
@@ -112,6 +112,140 @@ int main() {
 
 <Problems problems="match" />
 
+It is more than recommended to solve the first problem - Download Speed - in the section before jumping to this one.
```
Comment (SansPapyrus683) [inline content/6_Advanced/Max_Flow.mdx:115]:
```suggestion
It's recommended that you solve Download Speed before trying this one.
```

File: content/6_Advanced/Max_Flow.mdx [hunk 2]
```diff
@@ -112,6 +112,140 @@ int main() {
...
+</video>
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+const int INF = 1e9;
```
Comment (SansPapyrus683) [inline content/6_Advanced/Max_Flow.mdx:157]: int_max maybe?

File: content/6_Advanced/Max_Flow.mdx [hunk 3]
```diff
@@ -112,6 +112,140 @@ int main() {
...
+			}
+		}
+	}
+	// Returns true if an alternating path has been found
+	return dist[0] != INF;
+}
+
+// Returns true if an augmenting path has been found starting from vertex node
+bool dfs(int node) {
+	if (node == 0) return true;
```
Comment (SansPapyrus683) [inline content/6_Advanced/Max_Flow.mdx:196]: braces

File: content/6_Advanced/Max_Flow.mdx [hunk 4]
```diff
@@ -112,6 +112,140 @@ int main() {
...
+			Pair[node] = son;
+			Pair[son] = node;
+			return true;
+		}
+	}
+	dist[node] = INF;
+	return false;
+}
+
+int Hopcroft_Karp() {
```
Comment (SansPapyrus683) [inline content/6_Advanced/Max_Flow.mdx:208]: rename to `hopcroft_karp`

File: content/6_Advanced/Max_Flow.mdx [hunk 5]
```diff
@@ -112,6 +112,140 @@ int main() {
...
+	while (bfs()) {
+		for (int node = 1; node <= n; node++) {
+			// If node is unmatched but we can match it using an augmenting path
+			if (!Pair[node] && dfs(node)) { match++; }
+		}
+	}
+	return match;
+}
+
+// Hopcroft-Karp algorithm
```
Comment (SansPapyrus683) [inline content/6_Advanced/Max_Flow.mdx:220]:
```suggestion
```

File: content/6_Advanced/Max_Flow.mdx [hunk 6]
```diff
@@ -112,6 +112,140 @@ int main() {
...
+		g[x].push_back(y);
+		g[y].push_back(x);
+	}
+
+	cout << Hopcroft_Karp() << '\n';
+
+	for (int node = 1; node <= n; node++) {
+		if (Pair[node]) { cout << node << ' ' << Pair[node] - n << '\n'; }
+	}
+	return 0;
```
Comment (SansPapyrus683) [inline content/6_Advanced/Max_Flow.mdx:243]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/Max_Flow.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Max_Flow.mdx:157]: It's safer not tu put INT_MAX, could overflow

File: content/6_Advanced/Max_Flow.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Max_Flow.mdx:115]: Done

File: content/6_Advanced/Max_Flow.mdx (hunk 3 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Max_Flow.mdx:196]: Fair

File: content/6_Advanced/Max_Flow.mdx (hunk 5 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Max_Flow.mdx]: Ok

File: content/6_Advanced/Max_Flow.mdx (hunk 6 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Max_Flow.mdx:242]: Fair

File: content/6_Advanced/Max_Flow.mdx [hunk 7]
```diff
@@ -112,6 +112,140 @@ int main() {
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int INF = 1e9;
+
+int n, m, k;
+// dist[node] = the position of node in the alternating path
+vector<int> Pair, dist;
```
Comment (SansPapyrus683) [inline content/6_Advanced/Max_Flow.mdx:163]: can you rename `Pair` to something else so it isn't an eyesore

**Review by SansPapyrus683 — APPROVED** [review]
just one thing

**Review by ryanchou-dev — APPROVED** [review]
love it :)

File: content/6_Advanced/Max_Flow.mdx (hunk 7 above)
Comment (ryanchou-dev) [reply content/6_Advanced/Max_Flow.mdx:165]: bump

Comment (brebenelmihnea) [reply content/6_Advanced/Max_Flow.mdx]: Alright done

**Comment by SansPapyrus683** [comment]
uh can you rename the anim to hopcroft_karp for consistency?
i swear this will be the last thing

**Comment by brebenelmihnea** [comment]
@SansPapyrus683 alright done. You can merge it.

**Pull request merged**